tests/relay/schema.py

```diff
@@ -0,0 +1,209 @@
+from typing import Iterable, List, Optional
+from typing_extensions import Self
+
+import strawberry
+from strawberry.relay.utils import to_base64
+from strawberry.types import Info
+
+
+@strawberry.type
+class Fruit(strawberry.relay.Node):
+    _id: strawberry.Private[int]
+    name: str
+    color: str
+
+    @classmethod
+    def resolve_id(cls, root: Self, *, info: Optional[Info] = None):
```

Comment:
is there a reason why info is optional?

Reply:
I thought that someone might want to call this somewhere they don't have access to an `info` object, and objects usually don't require it to resolve their IDs. But it is still there for objects that might need them

I can remove the optional if you prefer and make it required

Reply:
+1 on making it non-optional. It's easy enough to simply call another method from this one if you need to share the implementation with another callee that doesn't have `Info`.

In my use-case, Info is required, and marking it Optional only adds to my implementation to ensure it's not None to satisfy type checking.